src/components/HomeLeaderboardPositionCard/HomeLeaderboardPositionCard.tsx

```diff
@@ -22,30 +35,121 @@ function LeaderboardUpDown({ direction, points }: LeaderboardUpDownProps) {
   );
 }
 
+function getPositionElement(position: number, name: string) {
```

Comment:
I am thinking how convenient would it be to extract this function, so that it can return cards according to the position, but also accepts another argument that determines if the card has this structure for "Teacher Home", or if it should follow the other "Full Leaderboard" structure
Another thought could be that this get extracted into a component that returns which icon should be displayed. I see you are repeating the name several times, maybe that could be used on upper function since that is dynamic and is not involved on the switch logic

Reply:
I believe that rendering multiple types of card would affect readability and maintenance as the function would become quite large. Same goes for icon.